### PR TITLE
Release v4.0.0.pre.alpha1

### PR DESCRIPTION
**Pull request by DirtyF**
👋 @jekyll/core [as planned](https://github.com/jekyll/jekyll/issues/7469) here is a first beta release of Jekyll v4.

- [x] Update documentation ([Preview](https://deploy-preview-7574--jekyllrb.netlify.com/docs/upgrading/3-to-4/))
- [x] Ask plugins authors to test `v4.0-alpha1`
  - [x] `jekyll-seo-tag` (https://github.com/jekyll/jekyll-seo-tag/pull/306)
  - [ ] `github-metadata` (https://github.com/jekyll/github-metadata/pull/163)
- [x] Release blog post ([Preview](https://deploy-preview-7574--jekyllrb.netlify.com/news/2019/03/18/jekyll-4-0-0-pre-alpha1-released/))
- [ ] Release gem

We'll release the gem after your feedback.

### CODE REVIEW

**Comment by ashmaroli** [comment]
@DirtyF I just realized that based on https://jekyllrb.com/docs/history/, it looks like there were never changes to the `History.markdown` for beta releases. Therefore, the "history" remains a "complete whole" when the actual version is shipped, instead of having to read through changes in all the "pre-release" history to know what changed.

In other words, simply linking to the current History doc in the release post for the pre-release should suffice (ref: https://jekyllrb.com/news/2015/01/24/jekyll-3-0-0-beta1-released/)

Also, following suit, can we have the beta released with a shorter name?
`4.0.0.beta` instead of `4.0.0.pre.beta1`?

Question: When do we publish and link to the "Upgrade to 4.0" document?

**Comment by ashmaroli** [comment]
@DirtyF Note: `jekyll-seo-tag` and the other plugins are **not broken due to the `Jekyll::Cache` class** (which is referred to as the Cache API)

**Comment by ashmaroli** [comment]
## Issue:
Plugins and Minima are not compatible with this beta (since they conservatively lock to `v3.x` :
https://travis-ci.org/jekyll/jekyll/jobs/507134523#L889

File: docs/_posts/2019-03-16-jekyll-4-0-0-pre-beta1-released.markdown [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+Jekyll now exposes a [caching API](/tutorials/cache-api/), that some plugins could benefit from. So if you're a plugin developer, this small tutorial is worth the read.
+
+Also be aware that it's a new *major* version, and it comes with a few breaking changes, notably :
+
+1. We dropped support for [Ruby 2.3 who goes EOL at the end of the month](https://www.ruby-lang.org/en/downloads/).
+   GitHub Pages runs Ruby 2.5.x, services lile Netlify or Forestry already upgraded to latest Ruby 2.6.x.
+2. `link` tag now include `relative_url` filter, hurray [no more need to prepend `{% raw %}{{ site.baseurl }}{% endraw %}` ](https://github.com/jekyll/jekyll/pull/6727).
+3. [`{% raw %}{% highlight %}{% endraw %}` now behaves like `{% raw %}{% raw %}{% endraw %}`](https://github.com/jekyll/jekyll/pull/6821), so you can no longer use `include` tags within.
+4. We dropped dependencies for Pygments, RedCarpet and rdiscount.
```
Comment (mattr-) [inline docs/_posts/2019-03-16-jekyll-4-0-0-pre-beta1-released.markdown]:
```suggestion
4. We dropped support for Pygments, RedCarpet and rdiscount.
```

**Comment by mattr-** [comment]
> ## Issue:
> Plugins and Minima are not compatible with this beta (since they conservatively lock to `v3.x` :
> https://travis-ci.org/jekyll/jekyll/jobs/507134523#L889

It's a pre-release. I think that's ok.

**Comment by ashmaroli** [comment]
> It's a pre-release. I think that's ok.

@mattr- In the release post we ask the users to test the beta out. But **they won't be able to** since
Bundler will not be able to resolve gems properly.
IMO we need to **relax the constraint** on in-house plugins and themes. For example:

```diff
# <plugin>.gemspec or <theme>.gemspec

- spec.add_runtime_dependency "jekyll", "~> 3.3"
+ spec.add_runtime_dependency "jekyll", ">= 3.3"
```

**Comment by mattr-** [comment]
Then let's make this an alpha release so that plugin authors can update their plugins and provide versions that will be compatible with both 3.8 and 4.x

**Comment by DirtyF** [comment]
renamed to alpha, releasing the gem after @mattr- re-review.

**Comment by DirtyF** [comment]
A test is now failing: https://travis-ci.org/jekyll/jekyll/jobs/508106997#L1040-L1044

**Comment by DirtyF** [comment]
@jekyllbot: merge +release

**Pull request merged**